src/compute/src/logging/compute.rs

```diff
@@ -310,6 +313,14 @@ pub(super) fn construct<A: Allocate + 'static>(
             .as_collection()
             .map(arrangement_heap_datum_to_row);
 
+        let error_count = error_count.as_collection().map(move |datum| {
+            Row::pack_slice(&[
+                Datum::String(&datum.export_id.to_string()),
+                Datum::UInt64(u64::cast_from(worker_id)),
+                Datum::Int64(datum.count),
```

Comment:
Can we move the `count` into the diff of the collection?

Reply:
It was like that before, but I changed it in response to the invalid retraction errors seen in CI: https://github.com/MaterializeInc/materialize/pull/21640#pullrequestreview-1625514879.

We assume that collections exported by compute always sum up to non-negative diffs. But per-worker error counts can be negative (as you pointed out elsewhere), so we can't encode them as diffs. Unless I'm missing something?

Reply:
Good point; we might be able to use negative counts if we have a `_RAW` collection that contains the negative counts, and a view on top that does a `COUNT(*)`, which should reveal the negative counts as data. Ofc, this might break down somewhere on the way.

Reply:
Yep, that was what the previous implementation was doing. The issue is, I think, that we provide a `_per_worker` view, which groups and sums counts per-worker, so the result can be negative.

We the the `_per_worker` view mainly for consistency (we have it for all the other introspection sources)`. We could skip it and only have a view that aggregates over all workers. This might work. However, we would be relying on error semantics and those are pretty weird right now. Also, we would still have queryable `_raw` collections that still can give you an error and that might still cause confusion down the road.

Reply:
Alright, let's leave it as-is. This is not a high-volume introspection source, so it should be OK. We could put a worker-local arrangement in front if it to avoid allocations that would cancel out, but you decide if it'd be worth it.

Reply:
I decided to not do this here as it seems inconsistent: Even in the worst case, the error count of a dataflow should update roughly as frequent as the dataflow frontier. But we don't do a pre-arrange for frontiers, so it's difficult to justify doing one for error counts. Maybe we should do a pre-arrange for both (and other logging arrangements too), but that's outside the scope of this PR.